File: demo-java/src/main/java/com/playwire/demo_java/ads/fullscreen/rewarded/RewardedActivity.java
package com.playwire.demo_java.ads.fullscreen.rewarded;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.intergi.playwiresdk.ads.fullscreen.PWFullScreenAd;
import com.intergi.playwiresdk.ads.fullscreen.rewarded.PWRewarded;
import com.playwire.demo_java.R;
import com.playwire.demo_java.misc.Constant;

public class RewardedActivity extends AppCompatActivity {
    PWRewarded rewarded;
    /// The ad unit name, e.g. 'banner-320x50', 'interstitial-home', 'rewarded-coins', etc.
    String adUnitName;

    TextView statusTextView;
    Button getRewardButton;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_rewarded);

        adUnitName = getIntent().getStringExtra(Constant.adUnitNameKey);
        statusTextView = findViewById(R.id.status_text_view);
        getRewardButton = findViewById(R.id.get_reward_button);
        getRewardButton.setEnabled(false);
        getRewardButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                showRewarded();
            }
        });

        loadRewarded();
    }

    void loadRewarded() {
        PWFullScreenAd.Listener listener = new PWFullScreenAd.Listener() {
            @Override
            public void onFullScreenAdLoaded(@NonNull PWFullScreenAd pwFullScreenAd) {
                statusTextView.setText(String.format("✅ The rewarded \"%s\" is loaded.", adUnitName));
                getRewardButton.setEnabled(true);
            }

            @Override
            public void onFullScreenAdFailedToLoad(@NonNull PWFullScreenAd pwFullScreenAd) {
                statusTextView.setText(String.format("❌ Failed to load the rewarded \"%s\".", adUnitName));
            }

            @Override
            public void onFullScreenAdShowedFullScreenContent(@NonNull PWFullScreenAd pwFullScreenAd) {
                statusTextView.setText(String.format("👍 The rewarded \"%s\" was successfully shown.", adUnitName));
                getRewardButton.setEnabled(false);
            }

            @Override
            public void onFullScreenAdFailedToShowFullScreenContent(@NonNull PWFullScreenAd pwFullScreenAd) {
                statusTextView.setText(String.format("❌ Failed to show the rewarded \"%s\".", adUnitName));
            }

            @Override
            public void onFullScreenAdDismissedFullScreenContent(@NonNull PWFullScreenAd pwFullScreenAd) {
            }

            @Override
            public void onFullScreenAdImpression(@NonNull PWFullScreenAd pwFullScreenAd) {
            }

            @Override
            public void onFullScreenAdReward(@NonNull PWFullScreenAd pwFullScreenAd, @NonNull String type, int amount) {
                /// Handle a reward regarding your business objectives.
                statusTextView.setText(String.format("🎉 The reward is earned.\n Type: %s \n Amount: %d", type, amount));
            }
        };

        rewarded = new PWRewarded(this, adUnitName, listener);

        /// Use `new PWLoadParams().withTargeting()` to pass your custom targets to ad request.
        /// PWLoadParams params = new PWLoadParams().withTargeting(new HashMap<String, String>() {{
        ///   put("age", "18-35");
        ///   put("page", "travel");
        /// }});
        /// this.rewarded.load(params);

        rewarded.load();
        statusTextView.setText(String.format("⏳ The rewarded \"%s\" is loading.", adUnitName));
    }

    void showRewarded() {
        if (rewarded == null || !rewarded.isLoaded()) {
            /// Load rewarded interstitial one more time or notify a user about error.
            loadRewarded();
            return;
        }
        rewarded.show();
    }
}

File: demo-kotlin/src/main/java/com/playwire/demo_kotlin/ads/fullscreen/rewarded/RewardedActivity.kt
package com.playwire.demo_kotlin.ads.fullscreen.rewarded

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.intergi.playwiresdk.ads.fullscreen.PWFullScreenAd
import com.intergi.playwiresdk.ads.fullscreen.rewarded.PWRewarded
import com.playwire.demo_kotlin.R
import com.playwire.demo_kotlin.misc.Constant

class RewardedActivity: AppCompatActivity() {
    // The ad unit name, e.g. 'banner-320x50', 'interstitial-home', 'rewarded-coins', etc.
    private lateinit var adUnitName: String

    private var rewarded: PWRewarded? = null

    private lateinit var statusTextView: TextView
    private lateinit var getRewardButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_rewarded)
        statusTextView = findViewById(R.id.status_text_view)
        getRewardButton = findViewById(R.id.get_reward_button)
        getRewardButton.setOnClickListener { showRewarded() }

        adUnitName = intent.getStringExtra(Constant.adUnitNameKey) ?: ""

        loadRewarded()
    }

    private fun loadRewarded() {
        val listener = object: PWFullScreenAd.Listener {

            override fun onFullScreenAdLoaded(ad: PWFullScreenAd) {
                statusTextView.text = "✅ The rewarded \"$adUnitName\" is loaded."
                getRewardButton.isEnabled = true
            }

            override fun onFullScreenAdFailedToLoad(ad: PWFullScreenAd) {
                statusTextView.text = "❌ Failed to load the rewarded \"$adUnitName\"."
            }

            override fun onFullScreenAdShowedFullScreenContent(ad: PWFullScreenAd) {
                statusTextView.text = "👍 The rewarded \"$adUnitName\" was successfully shown."
                getRewardButton.isEnabled = false
            }

            override fun onFullScreenAdReward(ad: PWFullScreenAd, type: String, amount: Int) {
                // Handle a reward regarding your business objectives.
                statusTextView.text = "🎉 The reward is earned.\n Type: $type \n Amount: $amount"
            }

            override fun onFullScreenAdFailedToShowFullScreenContent(ad: PWFullScreenAd) {
                statusTextView.text = "❌ Failed to show the rewarded \"$adUnitName\"."
            }

        }
        rewarded = PWRewarded(this, adUnitName, listener)
        rewarded?.load()

        // Use `PWLoadParams().withTargeting()` to pass your custom targets to ad request.
        // val params = PWLoadParams().withTargeting(
        //   mapOf(
        //      "age" to "18-32",
        //      "page" to "travel"
        //   )
        // )
        // rewarded?.load(params)

        getRewardButton.isEnabled = false
        statusTextView.text = "⏳ The rewarded \"$adUnitName\" is loading."
    }

    private fun showRewarded() {
        val rewarded = rewarded ?: return
        if (rewarded.isLoaded){
            rewarded.show()
        } else {
            // Load rewarded one more time or notify a user about error.
            loadRewarded()
        }
    }
}